lib/ui/sign_transaction/usecases/sign_transaction_use_case.dart

```diff
@@ -4,24 +4,23 @@ import 'package:hypha_wallet/core/error_handler/model/hypha_error.dart';
 import 'package:hypha_wallet/core/extension/scope_functions.dart';
 import 'package:hypha_wallet/core/network/api/eos_service.dart';
 import 'package:hypha_wallet/core/network/api/services/sign_transaction_callback_service.dart';
-import 'package:hypha_wallet/core/network/models/user_profile_data.dart';
-import 'package:hypha_wallet/core/shared_preferences/hypha_shared_prefs.dart';
+import 'package:hypha_wallet/core/network/repository/auth_repository.dart';
 import 'package:hypha_wallet/ui/architecture/interactor/base_usecase.dart';
 import 'package:hypha_wallet/ui/architecture/result/result.dart' as HResult;
 
 class SignTransactionUseCase extends InputUseCase<HResult.Result<String, HyphaError>, SignTransactionInput> {
   final EOSService eosService;
-  final HyphaSharedPrefs _appSharedPrefs;
+  final AuthRepository _authRepository;
   final SignTransactionCallbackService signTransactionCallbackService;
 
-  SignTransactionUseCase(this.eosService, this._appSharedPrefs, this.signTransactionCallbackService);
+  SignTransactionUseCase(this.eosService, this._authRepository, this.signTransactionCallbackService);
 
   @override
   Future<HResult.Result<String, HyphaError>> run(SignTransactionInput input) async {
-    final UserProfileData? userData = await _appSharedPrefs.getUserProfileData();
+    final userData = _authRepository.authDataOrCrash;
```

Comment:
I see now - I like the authDataOrCrash method!

-> userData should also have "network"

Then we can refactor and get rid of many network parameters... if the account we sign with is known, that account lives on a specific network. 

Of course accounts can have the same name, like joachim has the same name on EOS and Telos.